cpp/cmake_modules/Findutf8proc.cmake

```diff
@@ -15,10 +15,16 @@
 # specific language governing permissions and limitations
 # under the License.
 
+if(ARROW_UTF8PROC_USE_SHARED)
+  set(UTF8PROC_NAMES utf8proc)
+else()
+  set(UTF8PROC_NAMES utf8proc utf8proc_static)
+endif()
+
 if(utf8proc_ROOT)
   find_library(
     UTF8PROC_LIB
-    NAMES utf8proc
+    NAMES ${UTF8PROC_NAMES}
           "${CMAKE_SHARED_LIBRARY_PREFIX}utf8proc${CMAKE_SHARED_LIBRARY_SUFFIX}"
```

Comment:
Could you remove this?
It will find a shared library when we specify `-DARROW_UTFPROC8_USE_SHARED=NO`.